Enable all available layers

urish opened the pull request:
PR #28 with requested changes

fdb-git committed:
Update export.py

fdb-git committed:
Update prepare.py

kasbah commented on inkscape/svg2shenzhen/export.py:
```diff
@@ -457,6 +458,13 @@ def processExportLayer(self):
+                    mod = mod.replace('F.Cu', layer_label)
+                    open(layer_dest_kicad_path, "w").write(mod)
+
+
```
Can we not fix this in bitmap2component instead?

urish replied:
I'm not familiar with bitmap2component, but I can try to look into that. Are there any build instructions / pointers that can help here?

urish committed:
update bitmap2component to support arbitrary layer names

urish commented:
I updated the source code for bitmap2component so that any layer name can be used with it. I will test the new version tomorrow and let you know when I'm done

urish committed:
Make export.py backward compatible

Support both F.Silk/B.Silk and F.SilkS/B.SilkS to keep backward compatibility with older versions of svg2shenzhen.

Also remove Edge.Cuts from layer_map, has it has special handling in the exportEdgeCut() method.

urish commented:
Alright, tested and working for my project. I updated `export.py` as well to be backward compatible.

Do you want me to add the updated windows executable of `bitmap2component`  to this PR as well?

urish commented:
For future reference, this is how I built the the windows executable:

1. Downloaded [this msysgw distribution](https://nuwen.net/mingw.html) which includes the Boost library. Specifically, I downloaded `mingw-16.1-without-git.exe`.
2. Unzipped the file to `c:\MinGW` and added `c:\MinGW\bin` directory to my PATH.
3. Updated the `INCLUDES` line in the `Makefile` to read:
    ```
    INCLUDES=-Iinclude -Ipolygon -Ipotrace -Isrc -IC:\MinGW\include
    ```
4. Added the following at the top of `common/geometry/shape_poly_set.cpp`:
    ```
    #define _USE_MATH_DEFINES 
    #include <cmath>
    ```
5. Ran `make`, it compiled the project and created `bitmap2component.exe`
6. Ran `strip bitmap2component.exe` to make the file smaller (~4MB → 1.4MB, so it fits on a 💾)

badgeek commented:
@urish @kasbah thanks gonna check and merge this next sunday!

urish commented:
@badgeek 👍  had a chance to look at the PR?

badgeek commented:
@urish @kasbah @fabiodblab released 2.0.17 thanks all!

urish commented:
Thanks! I see the bitmap2component binaries haven't been updated in the repo. I published updated Windows / Linux binaries here: https://github.com/urish/svg2shenzhen/tree/enable-all-layers-bin/inkscape/svg2shenzhen

badgeek commented:
@urish updated the binaries + osx built thanks!
